File: examples/python/core/invoke/handle-invocation-error.py
from aws_durable_execution_sdk_python import (
    DurableContext,
    durable_execution,
    CallableRuntimeError,
)

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Handle errors from invoked functions."""
    try:
        result = context.invoke(
            function_name="risky-function",
            payload=event,
            name="risky_operation",
        )
        return {"status": "success", "result": result}
    
    except CallableRuntimeError as e:
        # Handle the error from the invoked function
        context.logger.error(f"Invoked function failed: {e}")
        return {
            "status": "failed",
            "error": str(e),
        }

File: examples/java/core/invoke/handle-invocation-error.java
public class HandleInvocationErrorHandler extends DurableHandler<Map<String, Object>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Handle errors from invoked functions
        try {
            var result = ctx.invoke(
                "risky_operation",
                "risky-function",
                event,
                Map.class
            );
            return Map.of("status", "success", "result", result);

        } catch (InvokeFailedException e) {
            // Handle the error from the invoked function
            ctx.getLogger().error("Invoked function failed: {}", e.getMessage());
            return Map.of(
                "status", "failed",
                "error", e.getMessage()
            );
        }
    }
}
